Sources/Datadog/CrashReporting/CrashReportingFeature.swift

```diff
@@ -0,0 +1,60 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Creates and owns componetns enabling Crash Reporting feature.
+internal final class CrashReportingFeature {
+    /// Single, shared instance of `CrashReportingFeature`.
+    static var instance: CrashReportingFeature?
+
+    /// Tells if the feature was enabled by the user in the SDK configuration.
+    static var isEnabled: Bool { instance != nil }
+
+    /// An interface for accessing the `DDCrashReportingPlugin` from `DatadogCrashReporting`.
+    let plugin: DDCrashReportingPluginType
+    /// Integration enabling sending crash reports as Logs. `nil` if the Logging feature is disabled.
+    let loggingIntegration: CrashReportingIntegration?
+    /// Integration enabling sending crash reports as RUM Errors. `nil` if the RUM feature is disabled.
+    let rumIntegration: CrashReportingIntegration?
+
+    init(
+        configuration: FeaturesConfiguration.CrashReporting,
+        loggingIntegration: CrashReportingIntegration?,
+        rumIntegration: CrashReportingIntegration?
+    ) {
+        self.plugin = configuration.crashReportingPlugin
+        self.loggingIntegration = loggingIntegration
+        self.rumIntegration = rumIntegration
+    }
+
+    func sendCrashReportIfFound() {
```

Comment:
That looks like an extra responsibility for this `Feature` class. Maybe extract that responsibility out to a dedicated class that is owned by `CrashReportingFeature`.

Reply:
Makes sense 👌, especially that with upcoming `RUMM-957` and `RUMM-1049` the crash reporting feature will learn another range of functionality: injecting SDK context into crash reports (last `RUMViewEvent`, last `UserInfo`, last `NetworkInfo` etc.).

I think it may be good to create `CrashReporter` class and put `sendCrashReportIfFound()` + upcoming ~`inject(context:)` in there. This way, the `Datadog.CrashReporter` will be abstracting the `CrashReporter` injected from `DatadogCrashReporting`. Seems good for testing, readability (single abstraction for 3rd party reporter) and will be unified with other features. WDYT?

Reply:
That sounds like a great plan!

Reply:
Done ✅ ([new commit](https://github.com/DataDog/dd-sdk-ios/pull/397/commits/71ed63cf793078ae8c44d054e8fe579224de5d48)).

I introduced the `CrashReporter` class which owns the `sendCrashReportIfFound()`. It will later expose an interface for injecting SDK context into crash reports.

It should be understood as an equivalent of `Logger` for `LoggingFeature`, `Tracer` for `TracingFeature` and `RUMMonitor` for `RUMFeature`. The only difference is that it remains `internal`, as we don't need to expose it for the user. I pinned it to `Global.crashReporter` as it has similar semantic as global `tracer` and `rum`.

The responsibility of `CrashReportingFeature` is now cleaned up. In next PRs it will be armed with necessary components for sending crash reports to Datadog.

// I also updated the diagram in the description of this PR.

Reply:
Wow this is great!